### PR TITLE
ci: split tests workflow into two and refactor them 

### PR DESCRIPTION
**Pull request by 170210**
# Description

This PR refactor `tests` workflow, in order to fix problem like: https://github.com/Finschia/cosmwasm/actions/runs/8165890934

This PR changed following:
- split `tests` workflow into two
- use matrix strategy to make contract test more clearly
- add missing ci tests for `floaty` and `cyberpunk`
- add document test for each contract in CI via `cargo test --doc` 
- use `cargo test --test integration` in CI

## Types of changes

- [x] Bug fix (changes which fixes an issue)
- [ ] New feature (changes which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] ETC (build, ci, docs, perf, refactor, style, test)

## Checklist

- [x] I followed the [contributing guidelines](https://github.com/Finschia/cosmwasm/blob/main/CONTRIBUTING.md).
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] The PR title and commits are followed [conventional commit form](https://www.conventionalcommits.org/en/v1.0.0).

### CODE REVIEW

File: .github/workflows/contract_tests.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            CHANGES_IN_REPO=$(git status --porcelain)
+            if [[ -n "$CHANGES_IN_REPO" ]]; then
+              echo "Repository is dirty. Showing 'git status' and 'git --no-pager diff' for debugging now:"
+              git status && git --no-pager diff
+              exit 1
+            fi
+      - name: Check formatting
+        working-directory: ${{env.working-directory}}
+        run: cargo fmt -- --check
+      - name: Clippy linting on burner
```
Comment (loloicci) [inline .github/workflows/contract_tests.yml:76]: on burner?

**Comment by loloicci** [comment]
Is there some example execution of this CI?

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -216,921 +216,34 @@ jobs:
...
-              exit 1
-            fi
-
-  fmt:
-    name: fmt
-    runs-on: ubuntu-latest
-    env:
-      root-directory: ./contracts
+      packages: ./packages
+      contracts: ./contracts
```
Comment (loloicci) [inline .github/workflows/tests.yml:224]: Is this needed?

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -216,921 +216,34 @@ jobs:
...
           target: wasm32-unknown-unknown
           profile: minimal
           override: true
+          components: rustfmt, clippy
       - name: Cache cargo
-        uses: actions/cache@v3
+        uses: actions/cache@v4
         with:
           path: ~/.cargo
           key: cargocache-v2-fmt-rust:1.60.0-${{ hashFiles('Cargo.lock', 'contracts/*/Cargo.lock') }}
```
[the remark this answers is not in the document]
Comment (170210) [reply .github/workflows/tests.yml:239]: Thanks, I will remove it.

**Review by kokeshiM0chi — APPROVED** [review]

**Review by loloicci — APPROVED** [review]

**Pull request merged**